pkg/controller/association/reconciler.go

```diff
@@ -258,6 +264,19 @@ func (r *Reconciler) reconcileAssociation(ctx context.Context, association commo
 		return commonv1.AssociationPending, err // maybe not created yet
 	}
 
+	secretsHash := fnv.New32a()
+	if r.AdditionalSecrets != nil {
+		additionalSecrets, err := r.AdditionalSecrets(ctx, r.Client, association)
+		if err != nil {
+			return commonv1.AssociationPending, err // maybe not created yet
+		}
+		for _, sec := range additionalSecrets {
+			if err := copySecret(ctx, r.Client, secretsHash, association.GetNamespace(), sec); err != nil {
```

Comment:
Are we supposed to watch this Secret? It's not reconciled if I delete it.

Reply:
You're right, it doesn't return, but the `agent-es` association controller is triggered on delete. I wonder if this:
https://github.com/naemono/cloud-on-k8s/blob/5dd7bb85550fd0d0b23af1762803025e725bf2d3/pkg/controller/association/dynamic_watches.go#L103-L117
is associating the secret incorrectly? I'll investigate and update.

Reply:
We were only watching the `source` secrets (the secrets we were copying into the target ns) and not the `target/created` secrets. It seems we would need to watch both:
1. if source secret changes, update target secret.
2. if target secret is changed/deleted, reconcile the secret.

I have updated and verified that both the target and source are re-created upon deletion.

Reply:
I noticed we have a similar issue with the fleet server es user/service token secret. Probably worth addressing in a separate PR though.

Reply:
I just realised we have captured my observation in an issue already https://github.com/elastic/cloud-on-k8s/issues/7170